### PR TITLE
[CDF-22879] Update extraction pipeline with NotificationConfig

### PR DESCRIPTION
**Pull request by doctrino**
## Description
Please describe the change you have made.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/data_classes/extractionpipelines.py [hunk 1]
```diff
@@ -224,6 +243,7 @@ class ExtractionPipelineWrite(ExtractionPipelineCore):
         metadata (dict[str, str] | None): Custom, application specific metadata. String key -> String value. Limits: Maximum length of key is 128 bytes, value 10240 bytes, up to 256 key-value pairs, of total size at most 10240.
         source (str | None): Source text value for extraction pipeline.
         documentation (str | None): Documentation text value for extraction pipeline.
+        notification_config (ExtractionPipelineNotificationConfiguration | None): Notification configuration for the extraction pipeline.
```
Comment (haakonvt) [inline cognite/client/data_classes/extractionpipelines.py:246]: Let's add in this bit of info from the API docs:

> Time in minutes to pass without any Run. Null if extraction pipeline is not checked.

File: cognite/client/data_classes/extractionpipelines.py [hunk 2]
```diff
@@ -203,6 +218,10 @@ def _load(cls, resource: dict, cognite_client: CogniteClient | None = None) -> E
                 ExtractionPipelineContact._load(contact) if isinstance(contact, dict) else contact
                 for contact in instance.contacts
             ]
+        if instance.notification_config and isinstance(instance.notification_config, dict):
```
Comment (haakonvt) [inline cognite/client/data_classes/extractionpipelines.py:221]: Can we skip the `isinstance` check on `dict`?
```suggestion
        if instance.notification_config:
```

**Review by haakonvt — APPROVED** [review]
I'd say just mark those tests as skip for now. You could add a condition to only skip for the next 7 or 14 days so that we don't forget about them 😄

File: cognite/client/data_classes/extractionpipelines.py (hunk 2 above)
Comment (doctrino) [reply cognite/client/data_classes/extractionpipelines.py:221]: No, then mypy complains. I don't like this pattern for loading, but did not what to do more changes than necessary.

**Pull request merged**